Sage/carbon
# feat: enable tree shaking - FE-4253

### Proposed behaviour

This PR changes the build process so that ES6 modules are now also provided along the old CommonJS ones

### Checklist




#### QA


### Testing instructions

1. Find a Sage repo that uses most recent version of `carbon-react` (sbc.accounting for instance)
2. Run build script from this repo (usually npm run build) and check the size of the built package (should be displayed in the terminal after succesful build)
3. After previous step install version of `carbon-react` based on build from this PR (there are instructions in the comment left by codesandbox bot, click second link from the top)
4. Run build again and compare the new size with the previous one (new one should be smaller)
5. Run unit tests in that repo to confirm that everything works.
